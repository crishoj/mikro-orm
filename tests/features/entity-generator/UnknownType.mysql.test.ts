import { MikroORM } from '@mikro-orm/mysql';
import { EntityGenerator } from '@mikro-orm/entity-generator';

const schema = `
  create table \`test\` (
    \`id\` bigint unsigned auto_increment,
    \`poly\` polygon,
    \`r\` bit(8) default b'1',
    primary key (\`id\`)
  );
`;

test('unknown-types', async () => {
  const orm = await MikroORM.init({
    dbName: 'unknown-types-test',
    port: 3308,
    discovery: { warnWhenNoEntities: false },
    extensions: [EntityGenerator],
    multipleStatements: true,
    ensureDatabase: false,
  });

  if (await orm.schema.ensureDatabase({ create: true })) {
    await orm.schema.execute(schema);
  }

  const dump = await orm.entityGenerator.generate();
  expect(dump).toMatchSnapshot();
  await orm.close(true);
});
