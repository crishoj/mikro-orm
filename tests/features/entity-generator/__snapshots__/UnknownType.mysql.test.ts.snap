// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown-types 1`] = `
[
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity()
export class Test {

  @PrimaryKey()
  id!: bigint;

  @Property({ columnType: 'polygon', nullable: true })
  poly?: unknown;

  @Property({ columnType: 'bit(8)', nullable: true, defaultRaw: \`b'1'\` })
  r?: unknown;

}
",
]
`;
